controllers/argocd/deployment.go

```diff
@@ -350,6 +343,55 @@ func isMergable(extraArgs []string, cmd []string) error {
 	return nil
 }
 
+// appendUniqueArgs appends extraArgs to cmd while ignoring any duplicate flags.
+func appendUniqueArgs(cmd []string, extraArgs []string) []string {
+	// Parse cmd into a map to track both flags and their associated values
+	existingArgs := make(map[string]string)
+	for i := 0; i < len(cmd); i++ {
+		arg := cmd[i]
+		if strings.HasPrefix(arg, "--") {
+			// Check if the next item is a value (not another flag)
+			if i+1 < len(cmd) && !strings.HasPrefix(cmd[i+1], "--") {
+				existingArgs[arg] = cmd[i+1] // Store flag-value pair
+				i++                          // Skip the value
+			} else {
+				existingArgs[arg] = "" // Store flag without value
+			}
+		}
+	}
+
+	// Iterate over extraArgs and append only if the flag and value do not already exist in cmd
+	for i := 0; i < len(extraArgs); i++ {
+		arg := extraArgs[i]
+		if strings.HasPrefix(arg, "--") {
+			// If this flag already exists in cmd, skip it
+			if _, exists := existingArgs[arg]; exists {
+				if i+1 < len(extraArgs) && !strings.HasPrefix(extraArgs[i+1], "--") {
+					i++ // Skip the associated value
```

Comment:
What if an existing flag is added with a new value, will that also be skipped?

Reply:
yes, it is skipped

Reply:
Is that the expected behavior, what if the user wants to change the current flag value?

Reply:
based on the existing unit test https://github.com/argoproj-labs/argocd-operator/blob/master/controllers/argocd/deployment_test.go#L1304  which is updating existing cmd i.e --redis command, I think this is expected behaviour, I might be wrong and open for suggestions

Reply:
Ok, maybe we can confirm with others as well. Otherwise looks good.

Reply:
I think users should be allowed to override default flags.